appcelerator/titanium_mobile
# fix(android): fall through request permissions

- Fall through `request<?>Permissions()`, removing the need for `has<?>Permissions()`

##### TEST CASE
```XML
<uses-permission android:name="android.permission.READ_CALENDAR"/>
<uses-permission android:name="android.permission.READ_CONTACTS"/>
<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
<uses-permission android:name="android.permission.CAMERA"/>
<uses-permission android:name="android.permission.RECORD_AUDIO"/>
```
```JS
const win = Ti.UI.createWindow({ backgroundColor: 'red' });

// Callback hell...
Ti.Calendar.requestCalendarPermissions(e => {
    if (e.success) {
        Ti.Contacts.requestContactsPermissions(e => {
            if (e.success) {
                Ti.Filesystem.requestStoragePermissions(e => {
                    if (e.success) {
                        Ti.Media.requestCameraPermissions(e => {
                            if (e.success) {
                                Ti.Media.requestAudioRecorderPermissions(e => {
                                    if (e.success) {
                                        win.backgroundColor = 'green';
                                    }
                                });
                            }
                        });
                    }
                });
            }
        });
    }
});

win.open();
```
- Open application and accept all permissions. Window should be green.
- Close application.
- Re-open application, permissions should not be requested and window should be green.

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-27741)
